go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -208,9 +212,9 @@ func (npw *nodePortWatcher) updateServiceFlowCache(service *kapi.Service, add, h
 							"actions=%s",
 							cookie, npw.ofportPhys, flowProtocol, svcPort.NodePort, actions),
 						// table=0, matches on return traffic from service nodePort and sends it out to primary node interface (br-ex)
-						fmt.Sprintf("cookie=%s, priority=110, in_port=%s, %s, tp_src=%d, "+
+						fmt.Sprintf("cookie=%s, priority=110, in_port=%s, dl_src=%s, %s, tp_src=%d, "+
 							"actions=output:%s",
-							cookie, npw.ofportPatch, flowProtocol, svcPort.NodePort, npw.ofportPhys)})
+							cookie, npw.ofportPatch, npw.ofm.defaultBridge.macAddress, flowProtocol, svcPort.NodePort, npw.ofportPhys)})
```

Comment:
@trozet don't we need locking for this value? `SetDefaultGatewayBridgeMAC()` locks it when writing.

Reply:
yeah, we are probably additionally missing correct locking elsewhere. I'll take a look.